datafusion/optimizer/tests/optimizer_integration.rs

```diff
@@ -294,6 +294,21 @@ fn eliminate_nested_filters() {
     assert_eq!(expected, format!("{plan:?}"));
 }
 
+#[test]
+fn eliminate_redundant_null_check() {
+    let sql = "\
+        SELECT col_int32, count(*) c
+        FROM test
+        GROUP BY col_int32
+        HAVING c IS NOT NULL";
+    let plan = test_sql(sql).unwrap();
+    let expected = "\
+          Projection: test.col_int32, count(*) AS c\
+        \n  Aggregate: groupBy=[[test.col_int32]], aggr=[[count(Int64(1)) AS count(*)]]\
+        \n    TableScan: test projection=[col_int32]";
+    assert_eq!(expected, format!("{plan:?}"));
```

Comment:
Note: `assert_eq!` names its parameters "left" and "right" and doesn't seem to designate which one is expected and which is actual. 
However, when a test is run from RustRover, the IDE is apparently opinionated and currently expects the "right" to be "the expected":

<img width="1156" alt="image" src="https://github.com/apache/datafusion/assets/144328/989d9c55-f032-49f0-880f-4bc93f672604">

I put the expected on the left, since this is what the existing code does.

Some RustRover references 

- https://github.com/intellij-rust/intellij-rust/issues/3814
- https://github.com/intellij-rust/intellij-rust/pull/3848

Reply:
filed as https://github.com/apache/datafusion/issues/11270